runner/runner.go

```diff
@@ -34,7 +34,7 @@ const (
 )
 
 func (p ProcessState) IsDone() bool {
-	return p == COMPLETE || p == FAILED || p == ABORTED || p == TIMEDOUT
+	return p == COMPLETE || p == FAILED || p == ABORTED || p == TIMEDOUT || p == BADREQUEST
```

Comment:
I'm kind of regretting BADREQUEST. I initially thought it would be a good idea in worker.thrift but didn't really want it in runner.go. We should consider converting it to an error as Dan said.